src/components/sidenav/sidenav.js

```diff
@@ -323,6 +326,77 @@ function SidenavDirective($mdMedia, $mdUtil, $mdConstant, $mdTheming, $animate,
       }
     }
 
+    function enableDragging() {
+      $mdGesture.register(element, 'drag', { horizontal: true });
+      element.on('$md.dragstart', onDragStart)
+        .on('$md.drag', onDrag)
+        .on('$md.dragend', onDragEnd);
+
+      var style = getComputedStyle(element[0]);
+      var sidenavWidth = parseInt(style.width);
```

Comment:
Should be better, I will test it
![](https://i.gyazo.com/b73de0f3756890467faed4765b77cbbb.png)